### DISCUSSION TITLE
Welcome to PDFGen Discussions!

### DISCUSSION THREAD

**Discussion by AndreRenaud** [AndreRenaud/PDFGen#113] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by WechatYouAndMe** [comment]
Do you have a simple way to create a pdf file with other language? Such as Japanese, Chinese....
If MCU can use PDFGen to generate PDF file, it's fantastic

**Reply by AndreRenaud** [reply]
Hi. Unfortunately PDFGen uses the built-in fonts from the PDF spec. These have a very limited character range (See appendix D in https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/pdfreference1.7old.pdf for the details). This rules out Japanese and Chinese. To support these we would need to add freetype font support. I have had a small attempt at this, but it is quite involved, and I have not completed the work.

At this stage, no, unfortunately it is not possible.

**Comment by nicola907** [comment]
Hi everyone. I am trying to add some text to my pdf. The function receives a c++ style string named text. The problem is that I don't know how to make a new line. I have tried using '\n', '\\n' and '\r\n'. Other suggestions?
```
string text = "a\nb";
pdf_add_text(pdf, NULL, text.c_str(), size, x, y, PDF_BLACK);//Here the character '\n' is not interpreted as new line
```

**Reply by AndreRenaud** [reply]
`pdf_add_text` unfortunately never does any wrapping for multiple lines. You'll need to use `pdf_add_text_wrap`, something like:

```
string text = "a\nb";
pdf_add_text_wrap(pdf, NULL, text.c_str(), size, x, y, 0, PDF_BLACK, 0, 200,  PDF_ALIGN_LEFT, NULL);
```
